src/Illuminate/Http/Concerns/InteractsWithInput.php

```diff
@@ -111,13 +111,24 @@ public function hasAny($keys)
      * Apply the callback if the request contains the given input item key.
      *
      * @param  string  $key
-     * @param  callable  $callback
+     * @param  callable|null  $callback
+     * @param  callable|null  $default
      * @return $this|mixed
      */
-    public function whenHas($key, callable $callback)
+    public function whenHas($key, callable $callback = null, callable $default = null)
     {
+        $value = data_get($this->all(), $key);
+
         if ($this->has($key)) {
-            return $callback(data_get($this->all(), $key)) ?: $this;
```

Comment:
You have changed the entire behavior of the function. Just add the default support. Don't change the behavior.